modules/tinymce/src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -1050,6 +1050,12 @@ function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}): DOMU
             return false;
           }
 
+          // Keep anchor in empty block
+          const isAnchor = name && !getAttrib(node, 'href') && (getAttrib(node, 'id') || getAttrib(node, 'name'));
```

Comment:
This name is rather misleading, as you're checking to see if it's a named anchor so I'd suggest calling this `isNamedAnchor` instead. Also this doesn't actually check if it's an anchor as you're just checking if name is truthy, so that should probably be `name === 'a'` as well.

- [ ] Rename variable and make sure we are actually matching an anchor